### PR TITLE
chore(components): Use @talend/utils

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
@talend/utils is now available.

**What is the chosen solution to this problem?**
Remove date helpers from @talend/components, use @talend/utils date helpers instead.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -7974,7 +7981,14 @@ dot-case@^3.0.4:
     no-case "^3.0.4"
     tslib "^2.0.3"
 
-dot-prop@^4.2.0, dot-prop@^5.1.0, dot-prop@^5.2.0:
+dot-prop@^4.2.0:
+  version "4.2.1"
```
Comment (jmfrancois) [inline yarn.lock:7985]: do you know why we have dot-prop twice in diff version ?

File: yarn.lock [hunk 2]
```diff
@@ -13643,7 +13662,17 @@ minimist-options@^3.0.1:
     arrify "^1.0.1"
     is-plain-obj "^1.1.0"
 
-minimist@0.0.8, minimist@1.1.x, minimist@^1.1.0, minimist@^1.1.1, minimist@^1.1.3, minimist@^1.2.0, minimist@^1.2.5:
+minimist@0.0.8:
```
Comment (jmfrancois) [inline yarn.lock:13665]: https://github.com/advisories/GHSA-vh95-rmgr-6w4m
You should enforce the usage of recent version

**Review by jmfrancois — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 1 above)
Comment (sgangnant) [reply yarn.lock]: Fixing `minimist` version addressed this 🍀

**Review by jmfrancois — APPROVED** [review]
👍

**Review by vbocquet — APPROVED** [review]

**Pull request merged**